test/src/unit-curl.cc

```diff
@@ -117,14 +117,11 @@ TEST_CASE(
   ThreadPool tp{1};
   ContextResources resources(
       cfg, tiledb::test::g_helper_logger(), 1, 1, "test");
-  tiledb::sm::RestClient rest_client;
-  REQUIRE(rest_client
-              .init(
-                  &tiledb::test::g_helper_stats,
-                  &cfg,
-                  &tp,
-                  tiledb::test::g_helper_logger(),
-                  resources)
-              .ok());
+  tiledb::sm::RestClient rest_client(
+      tiledb::test::g_helper_stats,
+      cfg,
+      tp,
+      *tiledb::test::g_helper_logger().get(),
+      std::move(resources.create_memory_tracker()));
```

Comment:
Can use `tiledb::test::create_test_memory_tracker()` here to resolve your CI failures